### PR TITLE
Coerce attention mask to bool datatype

### PR DESCRIPTION
**Pull request by jglaser**
A small fix that allows one to pass in int datatypes when bool is not supported
by the pytorch distributed environment

https://github.com/pytorch/pytorch/issues/32210

### CODE REVIEW

File: linear_mem_attention_pytorch/linear_mem_attn_torch.py [hunk 1]
```diff
@@ -30,6 +30,7 @@ def summarize_chunk(
         attn_weights = torch.einsum("bqhd,bkhd->bqhk", query, key)
         if mask is not None:
             max_neg = -torch.finfo(attn_weights.dtype).max
+            mask = mask.type(torch.bool)
```
Comment (hypnopump) [inline linear_mem_attention_pytorch/linear_mem_attn_torch.py]: better: 
`if mask.dtype != torch.bool: mask = mask.bool()`
so we can avoid a type conversion if not needed

**Comment by hypnopump** [comment]
pls adapt the change to the suggested one and i will auth merge

**Comment by jglaser** [comment]
> pls adapt the change to the suggested one and i will auth merge

done

**Comment by jglaser** [comment]
this is actually not needed. The pytorch distributed bug was triggered by my own model output, not this one.
